app/celery/v3/notification_tasks.py

```diff
@@ -183,23 +196,32 @@ def v3_send_email_notification(notification: Notification, template: Template):
     notification.sent_by = client.get_name()
     notification.reference = provider_reference
     db.session.commit()
+    return
 
 
-# TODO - retry conditions
-# TODO - error handling
-@notify_celery.task(serializer="pickle")
+@notify_celery.task(
+    serializer="pickle",
+    autoretry_for=(AwsPinpointException,),
+    retry_backoff=True,
+    retry_backoff_max=3600,  # 1 hour
+    max_retries=26           # This is about 1 day total of retries.
```

Comment:
Same regarding retries and `deliver_sms`.

Reply:
done